includes/MediaWiki/Api/BatchEditEndpoint.php

```diff
@@ -68,17 +68,16 @@ public function run() {
 			$otherItems = $request->otherItems();
 
 			$otherStatus = $this->persistItem( $reconciledItem, $otherItems );
+			$status->merge( $otherStatus );
 			if ( !$otherStatus->isOk() ) {
 				break;
 			}
 			$status->setResult( true, array_merge( $status->getValue(), [ $otherStatus->getValue() ] ) );
-			$status->merge( $otherStatus );
 		}
 
-		$response = [
-			'success' => $status->isOK(),
-		];
-		if ( $status->isOK() ) {
+		$response = $this->getResponseBody( $status );
+
+		if ( $status->isGood() ) {
```

Comment:
I’m not convinced by the change from `isOK()` to `isGood()` here… if the status contains a result, doesn’t it make sense to add it to the response, even if there are also errors?

Reply:
yeah this makes sense.